### PR TITLE
Add iast integration with GitHub

### PR DESCRIPTION
**Pull request by kecven**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](https://github.com/checkmarx-ltd/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/checkmarx-ltd/open-source-template/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Add integration IAST and CX-flow (github).

### References

https://github.com/checkmarx-ltd/cx-flow/pull/741 - Integration between IAST and CX-flow.

### Testing

src/test/resources/cucumber/features/integrationTests/cli/iastCli.feature

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 1401 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**1 Medium**<br />Low&nbsp;**1272 Low**<br />Info&nbsp;**128 Info**<br />
### Violation Summary
Medium&nbsp;**1** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1065264&projectid=3120)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[490](https://github.com/checkmarx-ltd/cx-flow/blob/pr-iast-2245-add_iast_integration_with_github/src/main/java/com/checkmarx/flow/CxFlowRunner.java#L490) |Medium|Use_of_System_exit|src/main/java/com/checkmarx/flow/CxFlowRunner.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1065264&projectid=3120&pathid=23)
***
Logo
**Checkmarx SCA - Scan Summary & Details**
### Cx-SCA Summary
**Total Packages Identified**: **269**
**Scan Risk Score**: **9.80**

High&nbsp;**11** **High** **severity vulnerabilities**
Medium&nbsp;**2** **Medium** **severity vulnerabilities**
Low&nbsp;**1** **Low** **severity vulnerabilities**
[View more details on Checkmarx UI](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4)
### Cx-SCA vulnerability result overview
|Vulnerability ID|Package|Severity|CVSS score|Publish date|Current version|Recommended version|Link in CxSCA|Reference – NVD link|
---|---|---|---|---|---|---|---|---
`CVE-2016-1000031`|commons-fileupload:commons-fileupload|HIGH|9.8|2016-10-25T14:29:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2016-1000031%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2016-1000031](https://nvd.nist.gov/vuln/detail/CVE-2016-1000031)
`Cxa9261daf-3755`|org.bouncycastle:bcprov-jdk15on|HIGH|9.8|2021-04-06T11:30:00|1.68|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/Cxa9261daf-3755%3AGradle-org.bouncycastle%3Abcprov-jdk15on-1.68/vulnerabilityDetails)|N\A
`CVE-2016-1000027`|org.springframework:spring-web|HIGH|9.8|2020-01-02T23:15:00|5.2.10.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2016-1000027%3AGradle-org.springframework%3Aspring-web-5.2.10.RELEASE/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2016-1000027`|org.springframework:spring-webmvc|HIGH|9.8|2020-01-02T23:15:00|5.2.10.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2016-1000027%3AGradle-org.springframework%3Aspring-webmvc-5.2.10.RELEASE/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2021-22118`|org.springframework:spring-web|HIGH|7.8|2021-05-27T15:15:00|5.2.10.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2021-22118%3AGradle-org.springframework%3Aspring-web-5.2.10.RELEASE/vulnerabilityDetails)|[CVE-2021-22118](https://nvd.nist.gov/vuln/detail/CVE-2021-22118)
`CVE-2014-0050`|commons-fileupload:commons-fileupload|HIGH|7.5|2014-04-01T06:27:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2014-0050%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2014-0050](https://nvd.nist.gov/vuln/detail/CVE-2014-0050)
`CVE-2016-3092`|commons-fileupload:commons-fileupload|HIGH|7.5|2016-07-04T22:59:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2016-3092%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2016-3092](https://nvd.nist.gov/vuln/detail/CVE-2016-3092)
`CVE-2013-2186`|commons-fileupload:commons-fileupload|HIGH|7.5|2013-10-28T21:55:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2013-2186%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2013-2186](https://nvd.nist.gov/vuln/detail/CVE-2013-2186)
`CVE-2021-25122`|org.apache.tomcat.embed:tomcat-embed-core|HIGH|7.5|2021-03-01T12:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2021-25122%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2021-25122](https://nvd.nist.gov/vuln/detail/CVE-2021-25122)
`CVE-2020-17527`|org.apache.tomcat.embed:tomcat-embed-core|HIGH|7.5|2020-12-03T19:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2020-17527%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2020-17527](https://nvd.nist.gov/vuln/detail/CVE-2020-17527)
`CVE-2021-25329`|org.apache.tomcat.embed:tomcat-embed-core|HIGH|7.0|2021-03-01T12:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2021-25329%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2021-25329](https://nvd.nist.gov/vuln/detail/CVE-2021-25329)
`CVE-2021-24122`|org.apache.tomcat.embed:tomcat-embed-core|MEDIUM|5.9|2021-01-14T15:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2021-24122%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2021-24122](https://nvd.nist.gov/vuln/detail/CVE-2021-24122)
`CVE-2020-17521`|org.codehaus.groovy:groovy|MEDIUM|5.5|2020-12-07T20:15:00|2.5.13|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2020-17521%3AGradle-org.codehaus.groovy%3Agroovy-2.5.13/vulnerabilityDetails)|[CVE-2020-17521](https://nvd.nist.gov/vuln/detail/CVE-2020-17521)
`CVE-2013-0248`|commons-fileupload:commons-fileupload|LOW|3.3|2013-03-15T20:55:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/01f73d10-3138-4139-a751-096732e81c3f/reports/6e9b0824-39e4-40ec-bcf7-a86dd646cbb4/vulnerabilities/CVE-2013-0248%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2013-0248](https://nvd.nist.gov/vuln/detail/CVE-2013-0248)

File: docs/CxIAST-Integration.md [hunk 1]
```diff
@@ -62,30 +62,56 @@ To ignore a severity, remove or comment that severity from the configuration fil
...
 
 ## <a name="bugTrackers">Bug Trackers</a>
-At present, CxFlow only supports Jira as a bug tracker when used with CxIAST.  
-Refer to [Jira Configuration](https://github.com/checkmarx-ltd/cx-flow/wiki/Bug-Trackers-and-Feedback-Channels#jira) for instructions on configuring CxFlow to work with Jira.
+
+At present, CxFlow only supports Jira and Github issue as a bug tracker when used with CxIAST.  
+Refer to [Jira Configuration](https://github.com/checkmarx-ltd/cx-flow/wiki/Bug-Trackers-and-Feedback-Channels#jira) for
+instructions on configuring CxFlow to work with Jira. Refer
+to [Github issue Configuration](https://github.com/checkmarx-ltd/cx-flow/wiki/Bug-Trackers-and-Feedback-Channels#github) for
+instructions on configuring CxFlow to work with Github issue.
```
Comment (yevgenykuz) [inline docs/CxIAST-Integration.md:77]: .... issue**s**.

File: docs/CxIAST-Integration.md [hunk 2]
```diff
@@ -62,30 +62,56 @@ To ignore a severity, remove or comment that severity from the configuration fil
...
+- The **assignee** field is set based on the `--assignee` argument that was passed to CxFlow, or based on the configured Jira
+  username. Refer to [Jira Configuration](https://github.com/checkmarx-ltd/cx-flow/wiki/Bug-Trackers-and-Feedback-Channels#jira)
+  for additional information.
+- The **description** field contains a link to the vulnerability in CxIAST Manager, scan tag, branch, repository name and severity
+  of vulnerability.
 
 An example for a Jira ticket is available here:  
 [[/Images/IAST2.png|Jira ticket example]]
 
+### Opening Tickets in Github issue
```
Comment (yevgenykuz) [inline docs/CxIAST-Integration.md:96]: Opening Github Issues

File: docs/CxIAST-Integration.md [hunk 3]
```diff
@@ -62,30 +62,56 @@ To ignore a severity, remove or comment that severity from the configuration fil
...
+
+CxFlow can open Github Issue tickets according to the CxIAST scan results.
```
Comment (yevgenykuz) [inline docs/CxIAST-Integration.md:98]: CxFlow can open Github issues according to the CxIAST scan results.

File: src/main/java/com/checkmarx/flow/controller/IastController.java [hunk 4]
```diff
@@ -39,20 +55,37 @@
                 .build());
     }
 
-    @PostMapping(value = {"/stop-scan-and-create-jira-issue/{scanTag}"})
+    @PostMapping(value = {"/stop-scan-and-create-{traker}-issue/{scanTag}"})
```
Comment (yevgenykuz) [inline src/main/java/com/checkmarx/flow/controller/IastController.java:58]: tracker

File: src/main/java/com/checkmarx/flow/controller/IastController.java [hunk 5]
```diff
@@ -39,20 +55,37 @@
...
     public ResponseEntity<EventResponse> stopScanAndCreateJiraIssue(
             @PathVariable(value = "scanTag", required = false) String scanTag,
-            @RequestHeader(value = TOKEN_HEADER) String token) {
+            @PathVariable(value = "traker", required = false) String bugTrackerName,
```
Comment (yevgenykuz) [inline src/main/java/com/checkmarx/flow/controller/IastController.java:61]: tracker

File: src/main/java/com/checkmarx/flow/service/IastService.java [hunk 6]
```diff
@@ -44,23 +42,28 @@
...
         this.helperService = helperService;
+        this.gitHubService = gitHubService;
 
         checkRequiredParameters();
 
-        severityToPriority.put(0, "Low");
-        severityToPriority.put(1, "Low");
-        severityToPriority.put(2, "Medium");
-        severityToPriority.put(3, "High");
+        jiraSeverityToPriority.put(Severity.INFO, "Low");
```
Comment (yevgenykuz) [inline src/main/java/com/checkmarx/flow/service/IastService.java:63]: Severity.INFO, "**INFO**"

**Pull request closed without merging**